org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/conversion/XbaseValueConverterService.java

```diff
@@ -251,4 +254,29 @@ public Integer toValue(String string, INode node) {
 			return super.toValue(withoutUnderscore, node);
 		}
 	}
+
+	@Inject
+	private WindowsEOLsAwareSTRINGValueConverter windowsEOLsAwareSTRINGValueConverter;
+
+	@Override
+	@ValueConverter(rule = "STRING")
+	public IValueConverter<String> STRING() {
+		return windowsEOLsAwareSTRINGValueConverter;
+	}
+
+	/**
+	 * Avoid Windows EOL characters from the original parsed text: this would
+	 * result in different generated Java files in Windows see
+	 * https://github.com/eclipse/xtext/issues/2293 This is aligned with Java
+	 * text blocks' "Normalization of Line Terminators"
+	 * 
+	 * @author Lorenzo Bettini - Initial contribution and API
+	 * @since 2.35
+	 */
+	public static class WindowsEOLsAwareSTRINGValueConverter extends STRINGValueConverter {
+		@Override
+		public String toValue(String string, INode node) {
+			return Strings.toUnixLineSeparator(super.toValue(string, node));
```

Comment:
It's an interesting case. I guess we want to treat an explicit `\r` in the string literal different from a regular windows newline. I've to think about it.

Reply:
cf
```java
	public static void main(String[] args) {
		String s = """
				some\r\nstring""";
		System.out.println(s);
		System.out.println(s.length());
		System.out.println('\r' == s.charAt(4));
	}
```

Reply:
@szarnekow a quick thought: I guess the original string should be normalized before passing it to super, not afterward as I'm doing now